File: java-maven/src/main/java/javamaven/javamaven/repository/HolidayRepository.java
package javamaven.javamaven.repository;

import javamaven.javamaven.entity.Holiday;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface HolidayRepository extends JpaRepository<Holiday, Long> {
    Optional<Holiday> findHolidayByName(String name);
}

File: kotlin-maven/src/main/kotlin/kotlinholidaytracker/kotlinholidaytracker/repository/HolidayRepository.kt
package kotlinholidaytracker.kotlinholidaytracker.repository

import kotlinholidaytracker.kotlinholidaytracker.entity.Holiday
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface HolidayRepository : JpaRepository<Holiday, Long> {
  fun findHolidayByName(name: String?): Optional<Holiday>
}